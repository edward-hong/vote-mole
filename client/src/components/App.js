import React from 'react'
import { BrowserRouter, Route } from 'react-router-dom'
import styled from 'styled-components'

import Home from './containers/Home'
import Navigation from './containers/Navigation'
import { HOME_PATH } from '../constants'

const Wrapper = styled.div`
	display: flex;
	min-height: 100vh;
	flex-direction: column;
`

const App = () => (
	<BrowserRouter>
		<Wrapper>
			<Navigation />
			<Route exact path={HOME_PATH} component={Home} />
		</Wrapper>
	</BrowserRouter>
)

export default App
