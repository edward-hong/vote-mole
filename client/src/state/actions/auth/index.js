import { FETCH_USER, FETCH_USER_FULFILLED } from '../../../constants'

export const fetchUser = () => ({ type: FETCH_USER })

export const fetchUserFulfilled = payload => ({
	type: FETCH_USER_FULFILLED,
	payload,
})
