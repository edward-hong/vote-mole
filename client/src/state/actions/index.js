export * from './auth'
export * from './polls'
