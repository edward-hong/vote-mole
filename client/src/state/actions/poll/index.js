import {
	POLL_INFO_TYPE,
	POLL_INFO_RECEIVED_TYPE,
	POLL_CLEAR_TYPE,
} from '../../../constants'

// Fetch poll
export const fetchPoll = payload => ({
	type: POLL_INFO_TYPE,
	payload,
})

// Dispatch received poll data to reducer
export const dispatchPollReceived = payload => ({
	type: POLL_INFO_RECEIVED_TYPE,
	payload,
})

// Action for clearing poll data from state
export const clearPoll = () => ({ type: POLL_CLEAR_TYPE })
