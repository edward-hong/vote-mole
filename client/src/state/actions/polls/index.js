import {
	POLL_ALL_TYPE,
	POLL_ALL_RECEIVED_TYPE,
	POLL_CLEAR_TYPE,
	POLL_USER_TYPE,
	POLL_USER_RECEIVED_TYPE,
} from '../../../constants'

// Fetch all polls
export const fetchAllPolls = payload => ({
	type: POLL_ALL_TYPE,
	payload,
})

// Dispatch received all polls to reducer
export const dispatchAllPollsReceived = payload => ({
	type: POLL_ALL_RECEIVED_TYPE,
	payload,
})

// Fetch user polls
export const fetchUserPolls = payload => ({
	type: POLL_USER_TYPE,
	payload,
})

// Dispatch received user polls to reducer
export const dispatchUserPollsReceived = payload => ({
	type: POLL_USER_RECEIVED_TYPE,
	payload,
})

// Clear polls state
export const clearPolls = () => ({ type: POLL_CLEAR_TYPE })
