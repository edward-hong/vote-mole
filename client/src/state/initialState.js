const initialState = {
	auth: null,
	poll: null,
}

export default initialState
