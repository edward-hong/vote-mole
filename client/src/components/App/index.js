import React, { useEffect } from 'react'
import { BrowserRouter, Route } from 'react-router-dom'
import { connect } from 'react-redux'
import styled from 'styled-components'

import Home from '../Home'
import Poll from '../Poll'
import Navigation from '../Navigation'
import Footer from '../Footer'
import { fetchUser } from '../../state/actions'
import { HOME_PATH, POLL_PATH } from '../../constants'

const Wrapper = styled.div`
	display: flex;
	min-height: 100vh;
	flex-direction: column;
`

const App = ({ fetchUser }) => {
	// Check whether a user is logged in or not
	useEffect(() => {
		fetchUser()
	}, [fetchUser])

	// Set frontend routes with Navigation and Footer appearing in all routes
	return (
		<BrowserRouter>
			<Wrapper>
				<Navigation />
				<Route exact path={HOME_PATH} component={Home} />
				<Route path={POLL_PATH} component={Poll} />
				<Footer />
			</Wrapper>
		</BrowserRouter>
	)
}

export default connect(
	null,
	{ fetchUser }
)(App)
