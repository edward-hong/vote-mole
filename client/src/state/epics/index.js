import { combineEpics } from 'redux-observable'
import { values } from 'ramda'

import * as authEpics from './auth'
import * as pollEpics from './polls'

const rootEpic = combineEpics(...values(authEpics), ...values(pollEpics))

export default rootEpic
