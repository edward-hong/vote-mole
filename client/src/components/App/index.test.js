import React from 'react'

import App from '.'
import renderWithRouterAndRedux from '../../utils/test/renderWithRouterAndRedux'
import { fetchUser, fetchAllPolls, clearPolls } from '../../state/actions'

jest.mock('../../state/actions', () => ({
	fetchUser: jest.fn(),
	fetchAllPolls: jest.fn(),
	clearPolls: jest.fn(),
}))

fetchUser.mockImplementation(() => ({ type: 'test' }))
fetchAllPolls.mockImplementation(() => ({ type: 'test' }))
clearPolls.mockImplementation(() => ({ type: 'test' }))

describe('App', () => {
	it('renders correctly', () => {
		const { container } = renderWithRouterAndRedux(<App />)

		expect(container.firstChild).toMatchSnapshot()
	})
})
