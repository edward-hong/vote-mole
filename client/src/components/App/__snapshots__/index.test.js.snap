// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
.c7 {
  position: relative;
  margin-top: 30px;
}

.c8 {
  margin: -24px 0 0 -24px;
  height: 49px;
  width: 49px;
  -webkit-animation: cssload-rotate 1150ms linear infinite;
  animation: cssload-rotate 1150ms linear infinite;
}

.c8,
.c8:before,
.c8:after {
  position: absolute;
  top: 50%;
  left: 50%;
  border: 1px solid #ccc;
  border-left-color: black;
  border-radius: 974px;
}

.c8:before {
  content: '';
  margin: -22px 0 0 -22px;
  height: 43px;
  width: 43px;
  -webkit-animation: cssload-rotate 1150ms linear infinite;
  animation: cssload-rotate 1150ms linear infinite;
}

.c8:after {
  content: '';
  margin: -28px 0 0 -28px;
  height: 55px;
  width: 55px;
  -webkit-animation: cssload-rotate 2300ms linear infinite;
  animation: cssload-rotate 2300ms linear infinite;
}

.c6 {
  margin-top: 1.5rem;
}

.c4 {
  text-align: center;
}

.c5 {
  text-align: center;
}

.c1 {
  margin-bottom: 2rem;
  border-bottom: 1px solid #21252a;
}

.c3 {
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.c2.c2 {
  color: #21252a;
  border-color: #21252a;
  width: 56px;
  height: 40px;
  position: relative;
  line-height: 50px;
}

.c2.c2 span {
  background-image: none;
  width: 30px;
  height: 2px;
  background-color: #21252a;
  position: absolute;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c2.c2 span:before,
.c2.c2 span:after {
  content: '';
  position: absolute;
  width: 30px;
  height: 2px;
  background-color: #21252a;
  left: -0.0001%;
  -webkit-transition: 0.2s;
  transition: 0.2s;
}

.c2.c2 span:before {
  top: -8px;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
}

.c2.c2 span:after {
  top: 8px;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
}

.c9 {
  text-align: center;
  margin: 3rem auto 1.5rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 100vh;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  class="c0"
>
  <nav
    class="c1 navbar navbar-expand-md navbar-light"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      <img
        alt="vote mole logo"
        src="https://res.cloudinary.com/avatarhzh/image/upload/v1509887327/build-a-voting-app/logo.svg"
      />
      VoteMole
    </a>
    <button
      active="0"
      class="c2 navbar-toggler"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="c3 collapse navbar-collapse"
    >
      <ul
        class="navbar-nav"
      />
    </div>
  </nav>
  <div
    class="container-fluid"
  >
    <h1
      class="c4"
    >
      VoteMole
    </h1>
    <h2
      class="c5"
    >
      Can You Dig It?
    </h2>
    <div
      class="c6 row"
    >
      <div
        class="col-12 col-sm-10 offset-sm-1 col-md-8 offset-md-2 col-lg-6 offset-lg-3"
      >
        <img
          alt="Vote Mole logo"
          class="logo-home"
          src="https://res.cloudinary.com/avatarhzh/image/upload/v1509887327/build-a-voting-app/logo.svg"
        />
      </div>
    </div>
    <div
      class="c6 row"
    >
      <div
        class="col-12 col-sm-10 offset-sm-1 col-md-8 offset-md-2 col-lg-6 offset-lg-3"
      >
        <div
          class="c7"
        >
          <div
            class="c8"
          />
        </div>
      </div>
    </div>
  </div>
  <footer
    class="c9"
  >
    Made by Edward Hong | 
    <a
      href="/privacy_policy"
    >
      Privacy Policy
    </a>
  </footer>
</div>
`;
